python/lsst/daf/butler/core/datasets/ref.py

```diff
@@ -52,6 +68,157 @@ class PositiveInt(ConstrainedInt):
     strict = True
 
 
+def _find_outside_stacklevel() -> int:
+    """Find the stacklevel for outside of lsst.daf.butler"""
+    stacklevel = 1
+    for i, s in enumerate(inspect.stack()):
+        module = inspect.getmodule(s.frame)
+        if module is None:
+            # Stack frames sometimes hang around so explicilty delete.
+            del s
+            continue
+        if not module.__name__.startswith("lsst.daf.butler"):
+            # 0 will be this function.
+            # 1 will be the caller
+            # and so does not need adjustment.
+            stacklevel = i
+            break
+        # Stack frames sometimes hang around so explicitly delete.
+        del s
```

Comment:
Maybe move `del s` right after `inspect.getmodule` so you don't need to do it twice?

Reply:
I probably should fix `utils` as well. Which is where this code came from. We can remove it when we remove the deprecation message.